tests/Application/Commands/DeleteStalePaymentDetailsTest.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Tests\Application\Commands;
+
+use MatchBot\Application\Commands\DeleteStalePaymentDetails;
+use MatchBot\Tests\Application\DonationTestDataTrait;
+use MatchBot\Tests\Application\StripeFormattingTrait;
+use MatchBot\Tests\TestCase;
+use Prophecy\Argument;
+use Prophecy\Prophecy\ObjectProphecy;
+use Psr\Log\NullLogger;
+use Stripe\Service\ChargeService;
+use Stripe\Service\CustomerService;
+use Stripe\Service\PaymentMethodService;
+use Stripe\StripeClient;
+use Symfony\Component\Console\Tester\CommandTester;
+use Symfony\Component\Lock\LockFactory;
+
+class DeleteStalePaymentDetailsTest extends TestCase
+{
+    use DonationTestDataTrait;
+    use StripeFormattingTrait;
+
+    public function testDeletingOneStaleCard(): void
+    {
+        $initDate = new \DateTimeImmutable('now');
+        $testCustomerId = 'cus_aaaaaaaaaaaa11';
+        $testPaymentMethodId = 'pm_aaaaaaaaaaaa13';
+        $testPaymentMethodFingerprint = 'Xt5EWLLDS7FJjR1c';
+
+        $stripeChargesProphecy = $this->prophesize(ChargeService::class);
+        $stripeChargesProphecy->search([
+            'query' => sprintf(
+                'customer:"%s" and payment_method_details.card.fingerprint:"%s" and status:"succeeded"',
+                $testCustomerId,
+                $testPaymentMethodFingerprint,
+            ),
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildEmptyCollection());
+
+        $stripePaymentMethodsProphecy = $this->prophesize(PaymentMethodService::class);
+        $stripePaymentMethodsProphecy->all([
+            'customer' => $testCustomerId,
+            'type' => 'card',
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture(
+                $this->getStripeHookMock('ApiResponse/pm'),
+            ));
+
+        // One PM should be detached i.e. soft deleted.
+        $stripePaymentMethodsProphecy->detach($testPaymentMethodId)
+            ->shouldBeCalledOnce()
+            ->willReturn($this->getStripeHookMock('ApiResponse/pm'));
```

Comment:
The code under test doesn't use the return value - any reason for providing it?
```suggestion
        $stripePaymentMethodsProphecy->detach($testPaymentMethodId)->shouldBeCalledOnce();
```

Reply:
My thinking was to match the return behaviour of the real library by default. I still think there's maybe some small value in that as protection against missed regressions from future changes in the code under test, but don't feel strongly about it.

Reply:
I feel like its better to make the test only as complex as required to document the behaviour of our code under test. Any changes in what the real library does or doesn't return can't break our code while our code doesn't use that return value.